### PR TITLE
Update ConfigContent to_dict functionality

### LINKED ISSUE #4028: ConfigContent.py does not correctly handles analysis SET_VAR
There is something buggy about  the following:
``` python 
user_config_content_as_dict = user_config_content.as_dict()
user_config_content_as_dict_simplified = {
    key: value[0][0] if len(value[0]) == 1 else value[0]
    for key, value in user_config_content_as_dict.items()
}
```

I think this logic should be part of `ConfigContent` and not added to `ResConfig` also the simplified dictionary does not account for keys that can have multiple elements like the following:

```
QUEUE_OPTION SLURM MAX_RUNNING 50
QUEUE_OPTION SLURM SBATCH /path/to/sbatch
QUEUE_OPTION SLURM SQUEUE /path/to/squeue
ANALYSIS_SET_VAR    STD_ENKF     ENKF_NCOMP    2
ANALYSIS_SET_VAR    ENKF_HIGH_TRUNCATION     ENKF_TRUNCATION    0.99
```

The following test should be able to pass:

```python 
def test_config_content_as_dict(tmpdir):
    with tmpdir.as_cwd():
        conf = ConfigParser()
        schema_item = conf.add("KEY_VALUE", False)
        schema_item.iset_type(0, ContentTypeEnum.CONFIG_INT)

        schema_item = conf.add("KEY", False)
        schema_item.iset_type(2, ContentTypeEnum.CONFIG_INT)

        with open("config", "w") as fileH:
            fileH.write("KEY_VALUE 42\n")
            fileH.write("KEY VALUE1 VALUE1 100\n")
            fileH.write("KEY VALUE2 VALUE2 200\n")
        content = conf.parse("config")
        content_as_dict = content.as_dict()
        user_config_content_as_dict_simplified = {
            key: value[0][0] if len(value[0]) == 1 else value[0]
            for key, value in content_as_dict.items()
        }
        assert user_config_content_as_dict_simplified == {'KEY': [['VALUE1', 'VALUE1', 100], ['VALUE2', 'VALUE2', 200]], 'KEY_VALUE': 42}
```

### PR DESCRIPTION
**Pull request by DanSava**
**Issue**
Resolves #4028

## Pre review checklist

- [ ] Added appropriate release note label
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Comment by DanSava** [comment]
Not sure why the `Run ert tests` for 3.10, 3.8 and 3.9 are hanging :confused:

**Review by valentin-krasontovitsch — APPROVED** [review]
I think this is good, but I'm sure @eivindjahren will want to have a look in particular at the test (edge) cases!

**Comment by valentin-krasontovitsch** [comment]
> Not sure why the `Run ert tests` for 3.10, 3.8 and 3.9 are hanging confused

i have the same issue on two PRs
i think github actions is broken - or something in the tests is seriously wrong

File: src/ert/_c_wrappers/enkf/ecl_config.py [hunk 1]
```diff
@@ -9,6 +9,15 @@
 from ert._c_wrappers.enkf.config_keys import ConfigKeys
 
 
+def get_value(config_dict_value: Optional[Union[list, str]]) -> Optional[str]:
```
Comment (eivindjahren) [inline src/ert/_c_wrappers/enkf/ecl_config.py:12]: `_get_value` ?

**Pull request merged**